docs/types.rst

```diff
@@ -918,16 +918,16 @@ Mappings
 ========
 
 You declare mapping types with the syntax ``mapping(_KeyType => _ValueType)``.
-The ``_KeyType`` can be any type except for a mapping, a dynamically sized
-byte array (a dynamically sized string array is valid), a contract, a function,
-an enum and a struct. ``_ValueType`` can be any type, including mappings.
+The ``_KeyType`` can be any elementary type except for a contracts, enums,
+structs or arrays, but dynamically-sized byte arrays and strings are fine.
+``_ValueType`` can be any type, including mappings.
 
 You can think of mappings as `hash tables <https://en.wikipedia.org/wiki/Hash_table>`_, which are virtually initialized
 such that every possible key exists and is mapped to a value whose
 byte-representation is all zeros, a type's :ref:`default value <default-value>`. The similarity ends there, the key data is not stored in a
 mapping, only its ``keccak256`` hash is used to look up the value.
 
-Because of this, mappings do not have a length or a concept of a "set" key or
+Because of this, mappings do not have a length or a concept of setting a key or
```

Comment:
"setting a key" and "a key being set" are two different things. You can of course set a key in the sense of setting a key to a value, but the key does not have the property of being set or unset.